mypy/build.py

```diff
@@ -806,7 +821,16 @@ def write_cache(id: str, path: str, tree: MypyFile,
     data_json_tmp = data_json + nonce
     meta_json_tmp = meta_json + nonce
     with open(data_json_tmp, 'w') as f:
-        json.dump(data, f, indent=2, sort_keys=True)
+        # TODO: The cache writing code currently assumes the data and meta
+        # file are always written together. However, with the new changes to
+        # incremental mode, we could instead write only an updated meta file
+        # and leave the data file alone when the public interface is unchanged.
```

Comment:
Hm, I don't think this assumption is made anywhere except right below where it adds the data file's mtime to the metadata. So writing only the metadata would be always have been safe, I think, as long as you write the data file first (since the meta file needs to include the correct mtime of the data file).